python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -2151,11 +2226,19 @@ def do_evaluate(
                 if name not in left.column_names_set
                 else f"{name}{suffix}",
                 left=False,
+                stream=stream,
             )
-            return DataFrame([*left_cols, *right_cols]).slice(zlice)
+            return DataFrame([*left_cols, *right_cols], stream=stream).slice(zlice)
         # TODO: Waiting on clarity based on https://github.com/pola-rs/polars/issues/17184
-        left_on = DataFrame(broadcast(*(e.evaluate(left) for e in left_on_exprs)))
-        right_on = DataFrame(broadcast(*(e.evaluate(right) for e in right_on_exprs)))
+        left_on = DataFrame(
+            broadcast(*(e.evaluate(left) for e in left_on_exprs)), stream=stream
+        )
+        right_on = DataFrame(
+            broadcast(*(e.evaluate(right) for e in right_on_exprs)), stream=stream
+        )
+        join_cuda_streams(
+            downstreams=(stream,), upstreams=(left_on.stream, right_on.stream)
+        )
```

Comment:
Sanity Check: Say your joining two tables from different streams on the the default stream, are there any concerns with races with this PR?

Reply:
Oh ok makes sense, thanks